mautic/mautic
# Campaign exclusion filters aren't working properly

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

The excluding filter in campaign 'condition' is not working correctly.
Behaviour of exclude/include operator for campaign condition Contact field value :-

Condition 1-

![image](https://github.com/user-attachments/assets/85a0d697-132e-48a4-bcda-9fdc6652b713)

**Positive path :-** Only contact who have value of field multiselect is a or b

**Negative path:-** All other contact who do not have any value or do not have value a and b for field multiselect

Condition 2-

![image](https://github.com/user-attachments/assets/5e1eb7e7-d600-41b3-b9f0-cf93eb566088)

**Positive path :-** All other contact who do not have any value or do not have value a and b for field multiselect

**Negative path:-** Only contact who have value of field multiselect is a or b






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create few contacts with various select/multi-select type fields
3. Create a segment to get all these contacts
4. Create a campaign with 'Contact field value' condition with filters on above field and a excluding operator with few values to exclude
5. Set any action after the condition
6. Execute campaign
7. Verify, Contacts with values which are excluded should not go through green path

**Test 2 -**

1. Create some contacts without assigning value to select/multi-select type field
2. Above contact should go in green path for exclude and in red path for include operator

**Test 3 -**

1. Verify same for company select/multi-select field
